addons/website_livechat/models/discuss_channel.py

```diff
@@ -58,8 +38,24 @@ def _field_store_repr(self, field_name):
 
     def _to_store_defaults(self, for_current_user=True):
         return super()._to_store_defaults(for_current_user=for_current_user) + [
+            Store.One(
+                "livechat_visitor_id",
+                [
+                    Store.One("country_id", ["code"]),
+                    "display_name",
+                    "history",
+                    "is_connected",
+                    Store.One("lang_id", ["name"]),
+                    Store.One(
+                        "partner_id",
+                        [Store.One("country_id", ["code"], rename="country"), "im_status"],
+                    ),
+                    Store.One("website_id", ["name"]),
+                ],
+                predicate=lambda channel: channel.livechat_visitor_id
```

Comment:
Note: for a follow up. This predicate will prevent from removing the visitor if it becomes unset.
We should be have a condition on channel type livechat maybe.